### PR TITLE
feat: Added Rooms

### PR DESCRIPTION
**Pull request by kristianbinau**
## Description

- Added Rooms
- Added /floor/{id}/rooms
- Added /room/type/{id}/rooms
## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Review by frederikpyt — APPROVED** [review]
LGTM!

File: src/modules/floor/floor.route.ts [hunk 1]
```diff
@@ -2,9 +2,10 @@ import { FastifyInstance, FastifyPluginOptions } from "fastify";
 import {
     createFloorHandler,
     browseFloorHandler,
```
Comment (Anders164a) [inline src/modules/floor/floor.route.ts]: Change browse to new scheme

File: src/modules/floor/floor.schema.ts [hunk 2]
```diff
@@ -34,40 +35,52 @@ const createFloorResponseSchema = z.object({
...
-    deleteFloorSchema,
-    deleteFloorResponseSchema
-}, {
-    $id: "floorSchema"
-});
+export const { schemas: floorSchemas, $ref } = buildJsonSchemas(
+    {
+        createFloorSchema,
+        createFloorResponseSchema,
+        browseFloorResponseSchema,
```
Comment (Anders164a) [inline src/modules/floor/floor.schema.ts]: Change to new naming scheme

File: src/modules/hotel/hotel.route.ts [hunk 3]
```diff
@@ -2,11 +2,11 @@ import { FastifyInstance, FastifyPluginOptions } from "fastify";
 import {
     createHotelHandler,
     browseHotelHandler,
```
Comment (Anders164a) [inline src/modules/hotel/hotel.route.ts]: Change to new naming scheme

File: src/modules/hotel/hotel.schema.ts [hunk 4]
```diff
@@ -4,30 +4,34 @@ import { floorSchema } from "../floor/floor.schema";
...
 
-const showHotelResponseSchema = z.object({
+const getHotelResponseSchema = z.object({
     id: z.number(),
-    ...hotelCore
+    ...hotelCore,
 });
 
-const browseHotelResponseSchema = z.array(showHotelResponseSchema);
+const browseHotelResponseSchema = z.array(getHotelResponseSchema);
```
Comment (Anders164a) [inline src/modules/hotel/hotel.schema.ts]: Change to new naming scheme

File: src/modules/hotel/hotel.schema.ts [hunk 5]
```diff
@@ -37,56 +41,61 @@ const createHotelResponseSchema = z.object({
...
+export type GetFloorsByHotelSchema = z.infer<typeof GetFloorsByHotelSchema>;
+export type GetRoomTypesByHotelSchema = z.infer<
+    typeof GetRoomTypesByHotelSchema
+>;
+
+export const { schemas: hotelSchemas, $ref } = buildJsonSchemas(
+    {
+        createHotelSchema,
+        createHotelResponseSchema,
+        browseHotelResponseSchema,
```
Comment (Anders164a) [inline src/modules/hotel/hotel.schema.ts]: Change to new naming scheme

File: src/modules/room/__test__/browseRoom.test.ts [hunk 6]
```diff
@@ -0,0 +1,112 @@
+import { FastifyInstance } from "fastify";
+import { build } from "../../../index";
+import { prisma } from "../../../plugins/prisma";
+
+describe("GET /api/room", () => {
```
Comment (Anders164a) [inline src/modules/room/__test__/browseRoom.test.ts:5]: Change file name to new naming scheme

File: src/modules/floor/__test__/showFloorRoom.test.ts [hunk 7]
```diff
@@ -0,0 +1,137 @@
+import { FastifyInstance } from "fastify";
+import { build } from "../../../index";
+import { prisma } from "../../../plugins/prisma";
+
+describe("GET /api/floor/:id/rooms", () => {
```
Comment (Anders164a) [inline src/modules/floor/__test__/showFloorRoom.test.ts:5]: Change file name to new naming scheme

File: src/modules/room/__test__/showRoom.test.ts [hunk 8]
```diff
@@ -0,0 +1,91 @@
+import { FastifyInstance } from "fastify";
+import { build } from "../../../index";
+import { prisma } from "../../../plugins/prisma";
+
+describe("GET /api/room/:id", () => {
```
Comment (Anders164a) [inline src/modules/room/__test__/showRoom.test.ts:5]: Change name of file to new naming scheme

File: src/modules/room/room.controller.ts [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+            CACHE_KEY_ROOM_TYPE_ROOMS + room.roomTypeId
+        );
+
+        return reply.code(201).send(room);
+    } catch (e) {
+        return reply.badRequest(errorMessage(e));
+    }
+}
+
+export async function browseRoomHandler(
```
Comment (Anders164a) [inline src/modules/room/room.controller.ts]: Change to new naming scheme

File: src/modules/room/room.route.ts [hunk 10]
```diff
@@ -0,0 +1,78 @@
+import { FastifyInstance, FastifyPluginOptions } from "fastify";
+import {
+    createRoomHandler,
+    browseRoomHandler,
```
Comment (Anders164a) [inline src/modules/room/room.route.ts]: Change to new naming scheme

File: src/modules/room/room.schema.ts [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+const getRoomSchema = z.object({
+    id: z.string(),
+});
+
+const getRoomResponseSchema = z.object({
+    id: z.number(),
+    ...roomCore,
+});
+
+const browseRoomResponseSchema = z.array(getRoomResponseSchema);
```
Comment (Anders164a) [inline src/modules/room/room.schema.ts]: Change to new naming scheme

File: src/modules/room/type/__test__/showRoomTypeRoom.test.ts [hunk 12]
```diff
@@ -0,0 +1,144 @@
+import { FastifyInstance } from "fastify";
+import { build } from "../../../../index";
+import { prisma } from "../../../../plugins/prisma";
+
+describe("GET /api/room/type/:id/rooms", () => {
```
Comment (Anders164a) [inline src/modules/room/type/__test__/showRoomTypeRoom.test.ts:5]: Change name of file new naming scheme

File: src/modules/room/type/type.route.ts [hunk 13]
```diff
@@ -2,9 +2,10 @@ import { FastifyInstance, FastifyPluginOptions } from "fastify";
 import {
     createRoomTypeHandler,
     browseRoomTypeHandler,
```
Comment (Anders164a) [inline src/modules/room/type/type.route.ts]: Change to new naming scheme

File: src/modules/room/type/type.schema.ts [hunk 14]
```diff
@@ -1,30 +1,33 @@
...
+        .string({
+            required_error: "Room type name is required",
+            invalid_type_error: "Name must be a string",
+        })
+        .min(1),
     description: z.string().nullable().optional(),
     size: z.string().nullable().optional(),
     price: z.number(),
     hotelId: z.number(),
 };
```
Comment (Anders164a) [inline src/modules/room/type/type.schema.ts:16]: We're missing people_amount, please add.. :)
image

File: src/modules/room/type/type.schema.ts [hunk 15]
```diff
@@ -1,30 +1,33 @@
...
 
-const showRoomTypeResponseSchema = z.object({
+const getRoomTypeResponseSchema = z.object({
     id: z.number(),
-    ...roomTypeCore
+    ...roomTypeCore,
 });
 
-const browseRoomTypeResponseSchema = z.array(showRoomTypeResponseSchema);
+const browseRoomTypeResponseSchema = z.array(getRoomTypeResponseSchema);
```
Comment (Anders164a) [inline src/modules/room/type/type.schema.ts]: Change to new naming scheme

**Review by Anders164a — CHANGES_REQUESTED** [review]
Rip 🤡

File: src/modules/room/type/__test__/createRoomType.test.ts [hunk 16]
```diff
@@ -157,4 +166,25 @@ describe("POST /api/room/type", () => {
             statusCode: 400,
         });
     });
+
+    it("should return status 400, when supportedPeople has not been provided", async () => {
+        const response = await fastify.inject({
+            method: "POST",
+            url: "/api/room/type",
+            payload: {
+                name: "",
```
Comment (Anders164a) [inline src/modules/room/type/__test__/createRoomType.test.ts:175]: There needs to be something, to make sure it tests the right property

Comment (Anders164a) [reply src/modules/room/type/__test__/createRoomType.test.ts:175]: I believe it's missing an ID too??

**Review by Anders164a — CHANGES_REQUESTED** [review]

**Review by Anders164a — APPROVED** [review]
🚀 🚗

**Pull request merged**